shared-module/epaperdisplay/EPaperDisplay.c

```diff
@@ -501,13 +505,45 @@ void epaperdisplay_epaperdisplay_collect_ptrs(epaperdisplay_epaperdisplay_obj_t
 }
 
 size_t maybe_refresh_epaperdisplay(void) {
-    for (uint8_t i = 0; i < CIRCUITPY_DISPLAY_LIMIT; i++) {
+    mp_int_t max_num_displays = CIRCUITPY_DISPLAY_LIMIT;
+
+    #if CIRCUITPY_OS_GETENV && CIRCUITPY_SET_DISPLAY_LIMIT
+    (void)common_hal_os_getenv_int("CIRCUITPY_DISPLAY_LIMIT", &max_num_displays);
+    #endif
+
+    for (uint8_t i = 0; i < max_num_displays; i++) {
+        #if CIRCUITPY_OS_GETENV && CIRCUITPY_SET_DISPLAY_LIMIT
+        if (i < CIRCUITPY_DISPLAY_LIMIT) {
+            if (displays[i].epaper_display.base.type != &epaperdisplay_epaperdisplay_type ||
+                displays[i].epaper_display.core.current_group != &circuitpython_splash) {
+                // Skip regular displays and those not showing the splash.
+                continue;
+            }
```

Comment:
Do this after you've gotten the display object to inspect. Maybe add an actual function to get display X that can switch been the static array and the dynamic one.

Reply:
> Do this after you've gotten the display object to inspect.

Thanks! moving it is much cleaner.

> Maybe add an actual function to get display X that can switch been the static array and the dynamic one.

When you suggest an actual function do you mean the macro functions I'm using in the other modules? I left it out here because there were only a couple instances and thought it was easier to understand without the macro. Or are you suggesting I add a new common_hal function that returns a separate display object from the appropriate memory structure and use that instead of the macros?